<?xml version="1.0"?>
<launch>
    <arg name="robot_name" default="sam"/>
    
    <!-- Node parameters -->
    <param name="robot_name" value="$(arg robot_name)"/>
    <param name="frame" value="map"/>

    <param name="path_name" value="/home/julian/catkin_ws/src/sam_slam/processing scripts/data/sim_testing"/>

    <!-- Data parameters -->
    <param name="dr_update_time" value="2.0"/>
    <param name="detect_update_time" value="0.5"/>

    <param name="camera_update_time" value="1.0"/>
    <param name="sss_update_time" value="2.0"/>
    <param name="sss_buffer_len" value="10"/>

    <!-- Detection and association parameters -->
    <param name="simulated_data" value="True"/>
    <param name="record_ground_truth" value="True"/>
    <param name="manual_associations" value="False"/>
    <param name="simulated_detections" value="True"/>
    <param name="prioritize_buoy_detections" value="False"/>

    <!-- Graph parameters -->
    <!-- Currently using default ISAM2 parameters -->
    <!-- Angles are in units of degrees, which are then converted to rads -->
    <!-- Prior sigmas -->
    <param name="prior_ang_sig_deg" value="2.5"/>
    <param name="prior_dist_sig" value="2.5"/>
    <!-- Buoy -->
    <param name="buoy_dist_sig" value="0.5"/>
    <!-- Odometry -->
    <param name="odo_ang_sig_deg" value=".5"/>
    <param name="odo_dist_sig" value=".1"/>
    <!-- Detection sigmas -->
    <param name="detect_ang_sig_deg" value="0.5"/>
    <param name="detect_dist_sig" value="0.1"/>

    <!-- Output parameters -->
    <param name="verbose_listener_DRs" value="false"/>
    <param name="verbose_listener_detections" value="true"/>
    <param name="verbose_listener_sonars" value="false"/>
    <param name="verbose_listener_buoys" value="true"/>
    <param name="verbose_listener_cameras" value="false"/>

    <param name="verbose_graph_update" value="false"/>
    <param name="verbose_graph_detections" value="true"/>

    <!-- NODES -->
    <!-- start rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sam_slam)/rviz/rviz_sim.rviz"/>

    <!-- SAM SLAM -->
    <node name="sam_slam_listener" pkg="sam_slam" type="sam_listener_online_slam_node.py" output="screen"/>

    <!-- Play the ROS bag -->
    <node name="rosbag_player" pkg="rosbag" type="play" args="-r 0.5 --clock /home/julian/sam_sim_1_edited.bag"/>

    <node name="sim_sss_object_detection" pkg="sss_object_detection" type="sim_sss_detection_publisher.py">
        <param name="robot_name" value="$(arg robot_name)"/>
    </node>

</launch>
